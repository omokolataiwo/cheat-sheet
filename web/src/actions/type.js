// Cheat Actions
export const RECEIVED_CATEGORIES = Symbol('RECEIVED_CATEGORIES');
export const INITIAL_CHEAT_STATE = Symbol('INITIAL_CHEAT_STATE');
export const FETCHING_CATEGORIES = Symbol('FETCHING_CATEGORIES');
export const SEARCH_COMPLETED = Symbol('SEARCH_COMPLETED');

// Cheat Favorite Actions
export const TOGGLE_LOVE_CHEAT = Symbol('TOGGLE_LOVE_CHEAT');
export const TOGGLE_LOVE_CHEAT_FAILURE = Symbol('TOGGLE_LOVE_CHEAT_FAILURE');

export const FETCHING_FAVORITE_CHEAT = Symbol('FETCHING_FAVORITE_CHEAT');
export const FETCH_FAVORITE_CHEAT_SUCCESS = Symbol('FETCH_FAVORITE_CHEAT_SUCCESS');
export const FETCH_FAVORITE_CHEAT_FAILURE = Symbol('FETCH_FAVORITE_CHEAT_FAILURE');

// User Actions
export const INITIAL_USER_STATE = Symbol('INITIAL_USER_STATE');
export const SIGNING_IN_USER = Symbol('SIGNING_IN_USER');
export const SIGNIN_SUCCESSFUL = Symbol('SIGNIN_SUCCESSFUL');
export const SIGNIN_FAILED = Symbol('SIGNIN_FAILED');
export const SIGNUP_SUCCESSFUL = Symbol('SIGNUP_SUCCESSFUL');
export const SIGNUP_FAILED = Symbol('SIGNUP_FAILED');
export const SIGNING_UP_USER = Symbol('SIGNING_UP_USER');
export const SIGNOUT_SUCCESSFULLY = Symbol('SIGNOUT_SUCCESSFULLY');
