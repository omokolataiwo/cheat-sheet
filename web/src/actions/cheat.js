import Axios from 'axios';
import {
  FETCHING_CATEGORIES,
  RECEIVED_CATEGORIES,
  TOGGLE_LOVE_CHEAT,
  TOGGLE_LOVE_CHEAT_FAILURE,
  FETCHING_FAVORITE_CHEAT,
  FETCH_FAVORITE_CHEAT_SUCCESS,
  FETCH_FAVORITE_CHEAT_FAILURE,
  SEARCH_COMPLETED
} from './type';
import { BASE_URL } from '../const';

const fetchingCategoies = () => ({ type: FETCHING_CATEGORIES });
const receivedCategories = categories => ({ type: RECEIVED_CATEGORIES, categories });

const toggleFavorite = (categoryId, cheatId) => ({ type: TOGGLE_LOVE_CHEAT, categoryId, cheatId });

const fetchingFavoriteCheat = () => ({ type: FETCHING_FAVORITE_CHEAT });
const fetchFavoriteCheatsSuccess = favorite => ({ type: FETCH_FAVORITE_CHEAT_SUCCESS, favorite });
const fetchFavoriteCheatsFailure = () => ({ type: FETCH_FAVORITE_CHEAT_FAILURE });

const searchCompleted = searchResult => ({ type: SEARCH_COMPLETED, searchResult });

export const fetchAllCategoriesWithCheats = () => (dispatch) => {
  const token = localStorage.getItem('token');

  if (token) {
    Axios.defaults.headers.common = { 'x-header-token': token };
  }

  dispatch(fetchingCategoies());
  return Axios.get(`${BASE_URL}/category`).then(({ data: { data } }) => {
    const { categories } = data;
    dispatch(receivedCategories(categories));
  });
};

const search = term => (category) => {
  const mCategory = { ...category };
  if (mCategory.title.toLowerCase().includes(term)) {
    return mCategory;
  }
  const cheats = mCategory.cheats.filter(
    cheat => cheat.description.toLowerCase().includes(term) || cheat.line.toLowerCase().includes(term)
  );
  mCategory.cheats = cheats;
  return mCategory;
};

export const searchSheetCheat = term => (dispatch, getState) => {
  if (term.length < 2) {
    return dispatch(searchCompleted(null));
  }

  const { categories } = getState().cheat;
  const searchResult = categories
    .map(search(term.toLowerCase()))
    .filter(category => !!category.cheats.length);
  return dispatch(searchCompleted(searchResult));
};

export const addToFavorite = cheatId => (dispatch, getState) => {
  const {
    user: { _id: userId }
  } = getState().user;

  return Axios.post(`${BASE_URL}/cheat/favorite`, { userId, cheatId })
    .then(({ data: { data: { category } } }) => {
      dispatch(toggleFavorite(category, cheatId));
    })
    .catch(() => dispatch({ type: TOGGLE_LOVE_CHEAT_FAILURE }));
};

export const fetchFavoriteCheats = () => (dispatch) => {
  Axios.defaults.headers.common = { 'x-header-token': localStorage.getItem('token') };
  dispatch(fetchingFavoriteCheat());
  return Axios.get(`${BASE_URL}/cheat/favorite`)
    .then(({ data: { data: cheats } }) => dispatch(fetchFavoriteCheatsSuccess(cheats)))
    .catch(() => dispatch(fetchFavoriteCheatsFailure()));
};
