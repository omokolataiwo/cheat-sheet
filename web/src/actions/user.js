import jwt from 'jsonwebtoken';
import Axios from 'axios';
import { BASE_URL } from '../const';
import {
  SIGNING_IN_USER,
  SIGNIN_SUCCESSFUL,
  SIGNIN_FAILED,
  INITIAL_USER_STATE,
  SIGNOUT_SUCCESSFULLY,
  SIGNUP_SUCCESSFUL,
  SIGNUP_FAILED,
  SIGNING_UP_USER
} from './type';

const signupSuccessful = user => ({ type: SIGNUP_SUCCESSFUL, user });
const signupFailed = error => ({ type: SIGNUP_FAILED, error });

const signinSuccessful = user => ({ type: SIGNIN_SUCCESSFUL, user });
const signinFailed = error => ({ type: SIGNIN_FAILED, error });
const logoutSuccessfully = () => ({ type: SIGNOUT_SUCCESSFULLY });

const handleSuccessfulUserAuthRequest = (data, dispatch, actionCreator) => {
  const { token } = data;
  localStorage.setItem('token', token);
  dispatch(actionCreator(jwt.decode(token)));
  dispatch({ type: INITIAL_USER_STATE });
};

const handleFailedUserAuthRequest = (response, dispatch, actionCreator) => {
  const {
    data: { error }
  } = response;
  dispatch(actionCreator(error));
};

export const signin = user => (dispatch) => {
  dispatch({ type: SIGNING_IN_USER });
  return Axios.post(`${BASE_URL}/user/signin`, { ...user })
    .then(({ data: { data } }) => {
      handleSuccessfulUserAuthRequest(data, dispatch, signinSuccessful);
    })
    .catch(({ response }) => {
      handleFailedUserAuthRequest(response, dispatch, signinFailed);
    });
};

export const signup = user => (dispatch) => {
  dispatch({ type: SIGNING_UP_USER });
  return Axios.post(`${BASE_URL}/user/`, { ...user })
    .then(({ data: { data } }) => {
      handleSuccessfulUserAuthRequest(data, dispatch, signupSuccessful);
    })
    .catch(({ response }) => {
      handleFailedUserAuthRequest(response, dispatch, signupFailed);
    });
};

export const logout = () => (dispatch) => {
  localStorage.removeItem('token');
  dispatch(logoutSuccessfully());
  dispatch({ type: INITIAL_USER_STATE });
};

export const resetUserState = () => (dispatch) => {
  dispatch({ type: INITIAL_USER_STATE });
};
