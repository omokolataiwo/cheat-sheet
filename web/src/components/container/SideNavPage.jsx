import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import SigninForm from './SigninForm';
import SignupForm from './SignupForm';
import { SIGNIN_SUCCESSFUL, SIGNUP_SUCCESSFUL } from '../../actions/type';

class SideNavPage extends React.Component {
  sideBarInstance = null;

  sNavInstance = null;

  modalInstance = null;

  componentDidMount() {
    const elem = document.querySelectorAll('.sidenav');
    const elModal = document.querySelectorAll('.modal');

    M.Sidenav.init(elem);
    this.sNavInstance = document.querySelector('#slide-out');
    this.sNavInstance = this.sNavInstance ? this.sNavInstance.M_Sidenav : null;
    M.Modal.init(elModal);
    this.modalInstance = document.querySelector('#signup-modal');
    this.modalInstance = this.modalInstance ? this.modalInstance.M_Modal : null;
  }

  componentDidUpdate() {
    const { userAction } = this.props;

    if (userAction.type === SIGNIN_SUCCESSFUL || userAction.type === SIGNUP_SUCCESSFUL) {
      this.sNavInstance.close();
      this.modalInstance.close();
    }
  }

  render() {
    if (localStorage.getItem('token')) {
      return null;
    }
    const {
      onFormFieldChange, onSignin, onSignup, user, errors, userAction
    } = this.props;
    return (
      <React.Fragment>
        <div id="signup-modal" className="modal">
          <div className="modal-content">
            {!localStorage.getItem('token') && (
              <React.Fragment>
                <SignupForm onFormFieldChange={onFormFieldChange} onSignup={onSignup} user={user} errors={errors.signup || {}} />
              </React.Fragment>
            )}
          </div>
        </div>

        <div id="slide-out" className="sidenav">
          <ul>
            <li className="signin-form">
              <SigninForm
                onFormFieldChange={onFormFieldChange}
                user={user}
                errors={errors.signin || {}}
                onSignin={onSignin}
                signInError={userAction.errors.signin}
              />
            </li>
          </ul>
          {'New User? '}
          <a className="modal-trigger" href="#signup-modal">
            Sign Up
          </a>
        </div>
        <a href="#" data-target="slide-out" className="sidenav-trigger">
          <FontAwesomeIcon icon="bars" />
        </a>
      </React.Fragment>
    );
  }
}

export default SideNavPage;
