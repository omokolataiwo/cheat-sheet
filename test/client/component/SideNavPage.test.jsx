import React from 'react';
import { shallow } from 'enzyme';
import SideNavPage from '../../../web/src/components/container/SideNavPage';
import { SIGNIN_SUCCESSFUL } from '../../../web/src/actions/type';

const props = {
  onFormFieldChange: jest.fn(() => {}),
  onSignin: jest.fn(() => {}),
  onSignup: jest.fn(() => {}),
  user: {},
  errors: {},
  userAction: { errors: {} }
};

const wrapper = shallow(<SideNavPage {...props} />);

describe('SideNavPage Component', () => {
  it('should render component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should close modal', () => {
    const instance = wrapper.instance();
    instance.sNavInstance = { close: () => {} };
    instance.modalInstance = { close: () => {} };
    wrapper.setProps({ ...props, userAction: { ...props.userAction, type: SIGNIN_SUCCESSFUL } });
  });
});
