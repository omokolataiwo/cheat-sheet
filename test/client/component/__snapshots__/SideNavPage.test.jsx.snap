// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNavPage Component should render component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SideNavPage
    errors={Object {}}
    onFormFieldChange={[MockFunction]}
    onSignin={[MockFunction]}
    onSignup={[MockFunction]}
    user={Object {}}
    userAction={
      Object {
        "errors": Object {},
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="modal"
          id="signup-modal"
        >
          <div
            className="modal-content"
          >
            <React.Fragment>
              <SignupForm
                errors={Object {}}
                onFormFieldChange={[MockFunction]}
                onSignup={[MockFunction]}
                user={Object {}}
              />
            </React.Fragment>
          </div>
        </div>,
        <div
          className="sidenav"
          id="slide-out"
        >
          <ul>
            <li
              className="signin-form"
            >
              <SigninForm
                errors={Object {}}
                onFormFieldChange={[MockFunction]}
                onSignin={[MockFunction]}
                user={Object {}}
              />
            </li>
          </ul>
          New User? 
          <a
            className="modal-trigger"
            href="#signup-modal"
          >
            Sign Up
          </a>
        </div>,
        <a
          className="sidenav-trigger"
          data-target="slide-out"
          href="#"
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon="bars"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
        </a>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="modal-content"
          >
            <React.Fragment>
              <SignupForm
                errors={Object {}}
                onFormFieldChange={[MockFunction]}
                onSignup={[MockFunction]}
                user={Object {}}
              />
            </React.Fragment>
          </div>,
          "className": "modal",
          "id": "signup-modal",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <React.Fragment>
              <SignupForm
                errors={Object {}}
                onFormFieldChange={[MockFunction]}
                onSignup={[MockFunction]}
                user={Object {}}
              />
            </React.Fragment>,
            "className": "modal-content",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <SignupForm
                errors={Object {}}
                onFormFieldChange={[MockFunction]}
                onSignup={[MockFunction]}
                user={Object {}}
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "errors": Object {},
                "onFormFieldChange": [MockFunction],
                "onSignup": [MockFunction],
                "user": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": Symbol(react.fragment),
          },
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ul>
              <li
                className="signin-form"
              >
                <SigninForm
                  errors={Object {}}
                  onFormFieldChange={[MockFunction]}
                  onSignin={[MockFunction]}
                  user={Object {}}
                />
              </li>
            </ul>,
            "New User? ",
            <a
              className="modal-trigger"
              href="#signup-modal"
            >
              Sign Up
            </a>,
          ],
          "className": "sidenav",
          "id": "slide-out",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <li
                className="signin-form"
              >
                <SigninForm
                  errors={Object {}}
                  onFormFieldChange={[MockFunction]}
                  onSignin={[MockFunction]}
                  user={Object {}}
                />
              </li>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <SigninForm
                  errors={Object {}}
                  onFormFieldChange={[MockFunction]}
                  onSignin={[MockFunction]}
                  user={Object {}}
                />,
                "className": "signin-form",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "errors": Object {},
                  "onFormFieldChange": [MockFunction],
                  "onSignin": [MockFunction],
                  "signInError": undefined,
                  "user": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "li",
            },
            "type": "ul",
          },
          "New User? ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign Up",
              "className": "modal-trigger",
              "href": "#signup-modal",
            },
            "ref": null,
            "rendered": "Sign Up",
            "type": "a",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon="bars"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />,
          "className": "sidenav-trigger",
          "data-target": "slide-out",
          "href": "#",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "border": false,
            "className": "",
            "fixedWidth": false,
            "flip": null,
            "icon": "bars",
            "inverse": false,
            "listItem": false,
            "mask": null,
            "pull": null,
            "pulse": false,
            "rotation": null,
            "size": null,
            "spin": false,
            "symbol": false,
            "title": "",
            "transform": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "a",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="modal"
            id="signup-modal"
          >
            <div
              className="modal-content"
            >
              <React.Fragment>
                <SignupForm
                  errors={Object {}}
                  onFormFieldChange={[MockFunction]}
                  onSignup={[MockFunction]}
                  user={Object {}}
                />
              </React.Fragment>
            </div>
          </div>,
          <div
            className="sidenav"
            id="slide-out"
          >
            <ul>
              <li
                className="signin-form"
              >
                <SigninForm
                  errors={Object {}}
                  onFormFieldChange={[MockFunction]}
                  onSignin={[MockFunction]}
                  user={Object {}}
                />
              </li>
            </ul>
            New User? 
            <a
              className="modal-trigger"
              href="#signup-modal"
            >
              Sign Up
            </a>
          </div>,
          <a
            className="sidenav-trigger"
            data-target="slide-out"
            href="#"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon="bars"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
          </a>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="modal-content"
            >
              <React.Fragment>
                <SignupForm
                  errors={Object {}}
                  onFormFieldChange={[MockFunction]}
                  onSignup={[MockFunction]}
                  user={Object {}}
                />
              </React.Fragment>
            </div>,
            "className": "modal",
            "id": "signup-modal",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <React.Fragment>
                <SignupForm
                  errors={Object {}}
                  onFormFieldChange={[MockFunction]}
                  onSignup={[MockFunction]}
                  user={Object {}}
                />
              </React.Fragment>,
              "className": "modal-content",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <SignupForm
                  errors={Object {}}
                  onFormFieldChange={[MockFunction]}
                  onSignup={[MockFunction]}
                  user={Object {}}
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "errors": Object {},
                  "onFormFieldChange": [MockFunction],
                  "onSignup": [MockFunction],
                  "user": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": Symbol(react.fragment),
            },
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ul>
                <li
                  className="signin-form"
                >
                  <SigninForm
                    errors={Object {}}
                    onFormFieldChange={[MockFunction]}
                    onSignin={[MockFunction]}
                    user={Object {}}
                  />
                </li>
              </ul>,
              "New User? ",
              <a
                className="modal-trigger"
                href="#signup-modal"
              >
                Sign Up
              </a>,
            ],
            "className": "sidenav",
            "id": "slide-out",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <li
                  className="signin-form"
                >
                  <SigninForm
                    errors={Object {}}
                    onFormFieldChange={[MockFunction]}
                    onSignin={[MockFunction]}
                    user={Object {}}
                  />
                </li>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <SigninForm
                    errors={Object {}}
                    onFormFieldChange={[MockFunction]}
                    onSignin={[MockFunction]}
                    user={Object {}}
                  />,
                  "className": "signin-form",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "errors": Object {},
                    "onFormFieldChange": [MockFunction],
                    "onSignin": [MockFunction],
                    "signInError": undefined,
                    "user": Object {},
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "li",
              },
              "type": "ul",
            },
            "New User? ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign Up",
                "className": "modal-trigger",
                "href": "#signup-modal",
              },
              "ref": null,
              "rendered": "Sign Up",
              "type": "a",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon="bars"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />,
            "className": "sidenav-trigger",
            "data-target": "slide-out",
            "href": "#",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "border": false,
              "className": "",
              "fixedWidth": false,
              "flip": null,
              "icon": "bars",
              "inverse": false,
              "listItem": false,
              "mask": null,
              "pull": null,
              "pulse": false,
              "rotation": null,
              "size": null,
              "spin": false,
              "symbol": false,
              "title": "",
              "transform": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "a",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
